modules/reactor/src/meshgenerators/PolygonMeshGeneratorBase.C

```diff
@@ -0,0 +1,1229 @@
+
+        for (unsigned int j = 1; j <= angle_number; j++)
+        {
+          (*node_id)++;
+          const Real cell_boundary_p_x =
+              corner_p[0][0] + (corner_p[1][0] - corner_p[0][0]) *
+                                   (azimuthal_tangent.size() == 0 ? ((Real)j / (Real)angle_number)
+                                                                  : (azimuthal_tangent[j] / 2.0));
+          const Real cell_boundary_p_y =
+              corner_p[0][1] + (corner_p[1][1] - corner_p[0][1]) *
+                                   (azimuthal_tangent.size() == 0 ? ((Real)j / (Real)angle_number)
+                                                                  : (azimuthal_tangent[j] / 2.0));
+          // cell_boundary_p(s) are the points on the cell's six boundries (flat sides) at
+          // differnt azimuthal angles
+          const Real pin_azimuthal_p_x =
+              cell_boundary_p_x * bin_radial_distance / (0.5 * corner_to_corner);
+          const Real pin_azimuthal_p_y =
+              cell_boundary_p_y * bin_radial_distance / (0.5 * corner_to_corner);
+
+          // pin_azimuthal_p are the points on the bins towards differnt azimuthal angles, at
+          // differnt intervals; excluding the ones produced by pin_corner_p
+          Point p = Point(pin_azimuthal_p_x, pin_azimuthal_p_y, 0.0);
+          (*nodes)[*node_id] = mesh->add_point(p, *node_id);
+        }
+      }
+    }
+  }
+  return dynamic_pointer_cast<ReplicatedMesh>(mesh);
+}
+
+// 'cen_quad_elem_def' defines quad elements in the very central region of the polygon.
+std::unique_ptr<ReplicatedMesh>
+PolygonMeshGeneratorBase::cen_quad_elem_def(std::unique_ptr<ReplicatedMesh> mesh,
+                                            const unsigned int div_num,
+                                            const std::vector<Node *> nodes,
+                                            const unsigned int block_id_shift,
+                                            const unsigned int boundary_id_shift,
+                                            std::vector<std::vector<dof_id_type>> * id_array)
+{
+
+  BoundaryInfo & boundary_info = mesh->get_boundary_info();
+
+  // This loop defines quad elements for the central regions except for the outmost layer
+  for (unsigned int i = 0; i < div_num - 1; i++)
+  {
+    unsigned int id_x = 0;
+    unsigned int id_y = i;
+    for (unsigned int j = 0; j < 2 * i + 1; j++)
+    {
+      Elem * elem_Quad4 = mesh->add_elem(new Quad4);
+      elem_Quad4->set_node(0) = nodes[(*id_array)[id_x][id_y]];
+      elem_Quad4->set_node(3) = nodes[(*id_array)[id_x][id_y + 1]];
+      elem_Quad4->set_node(2) = nodes[(*id_array)[id_x + 1][id_y + 1]];
+      elem_Quad4->set_node(1) = nodes[(*id_array)[id_x + 1][id_y]];
+      elem_Quad4->subdomain_id() = 1 + block_id_shift;
+      if (id_x == 0)
+        boundary_info.add_side(elem_Quad4, 3, SLICE_BEGIN);
+      if (id_y == 0)
+        boundary_info.add_side(elem_Quad4, 0, SLICE_END);
```

Comment:
what are slice begin and slice end for? They cut match but have different normals right?
Add a comment maybe

Reply:
This just facilitate the stitching of these slices into a polygon. Will add a comment.